src/lib/modules/contracts_manager/index.js

```diff
@@ -293,7 +293,8 @@ class ContractsManager {
       },
       function compileContracts(callback) {
         self.events.emit("status", __("Compiling..."));
-        if (self.compileOnceOnly && self.compiledContracts && Object.keys(self.compiledContracts).length) {
+        const everyFileWasCompiled = self.compiledContracts && self.contractsFiles && self.contractsFiles.every(contractFile => Object.values(self.compiledContracts).find(contract => contract.originalFilename === contractFile.filename));
```

Comment:
I think it'd be good to put this in a method `allContractsCompiled()` or sth. and call that here, just for readability-sake.

Reply:
Updated.

Reply:
As @jrainville pointed out, I was more looking for sth. like: https://github.com/embark-framework/embark/pull/1254#discussion_r248282571